src/hooks/useNotificationManager.tsx

```diff
@@ -0,0 +1,147 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { AppState, AppStateStatus } from 'react-native';
+import {
+  PrivatePostNotification,
+  useNotifications,
+  Notification,
+} from './useNotifications';
+import { useAsyncStorage } from './useAsyncStorage';
+import { onNotificationReceived } from '../common';
+import { compact } from 'lodash';
+
+export interface NotificationsManagerContextType {
+  unreadCount: number | undefined;
+  setNotificationsRead: () => void;
+  /**
+   * undefined - async storage has not been loaded yet
+   * null - async storage has been loaded but no value
+   * Date - async storage has loaded and key was set
+   */
+  lastReadAt: Date | null | undefined;
+  privatePostsUserIds: string[] | undefined;
+}
+
+export const NotificationsManagerContext =
+  createContext<NotificationsManagerContextType>({
+    unreadCount: 0,
+    privatePostsUserIds: [],
+    setNotificationsRead: () => {},
+    lastReadAt: null,
+  });
+
+export const NotificationsManagerProvider = ({
```

Comment:
This name seems a little vague since this provider deals with private messages directly. Is the thinking that this would be expanded to other types so that it is responsible for managing all types of notifications and tracking the last read information?

Reply:
This was the original name given from where I copied this code over from. The idea is that this will eventually be more for the notifications tab and tracking new notifications from the feed rather than used for private messaging but I thought it was OK to do it this way for a first pass.